### PR TITLE
make `FutureActorRef<T>` unsealed

### PR DESCRIPTION
**Pull request by Aaronontheweb**
## Changes

Made `FutureActorRef<T>` unsealed in order to resolve https://github.com/petabridge/phobos-issues/issues/74 - working on testing this to verify that is actually does work.

## Checklist

For significant changes, please ensure that the following have been completed (delete if not relevant):

* [x] This change follows the [Akka.NET API Compatibility Guidelines](https://getakka.net/community/contributing/api-changes-compatibility.html).
* [x] This change follows the [Akka.NET Wire Compatibility Guidelines](https://getakka.net/community/contributing/wire-compatibility.html).
* [x] I have [reviewed my own pull request](https://getakka.net/community/contributing/index.html#review-your-own-pull-requests).
* [x] Changes in public API reviewed, if any.

### CODE REVIEW

File: src/core/Akka.API.Tests/CoreAPISpec.ApproveCore.verified.txt [hunk 1]
```diff
@@ -885,7 +885,7 @@ namespace Akka.Actor
         public Failures() { }
         public System.Collections.Generic.List<Akka.Actor.Failure> Entries { get; }
     }
-    public sealed class FutureActorRef<T> : Akka.Actor.MinimalActorRef
```
Comment (Aaronontheweb) [inline src/core/Akka.API.Tests/CoreAPISpec.ApproveCore.verified.txt:888]: Not a breaking change

File: src/core/Akka/Actor/ActorRef.cs [hunk 2]
```diff
@@ -122,9 +122,12 @@ protected override void TellInternal(object message, IActorRef sender)
                     handled = _result.TrySetException(f.Cause
                         ?? new TaskCanceledException("Task cancelled by actor via Failure message."));
                     break;
+#pragma warning disable CS0618
```
Comment (Aaronontheweb) [inline src/core/Akka/Actor/ActorRef.cs:125]: Disabled a compiler warning here for a backwards-compat setting

File: src/core/Akka/Actor/Futures.cs [hunk 3]
```diff
@@ -164,7 +164,7 @@ public static Task<T> Ask<T>(this ICanTell self, Func<IActorRef, object> message
             //The future actor needs to be registered in the temp container
             provider.RegisterTempActor(future, path);
             var message = messageFactory(future);
-            self.Tell(message, future);
+            future.DeliverAsk(message, self);
```
Comment (Aaronontheweb) [inline src/core/Akka/Actor/Futures.cs:167]: Made the `.Tell` implementation internal to the `FutureActorRef<T>` so it's possible to generate telemetry here (and do things like filter which `Ask<T>` operations should be traced and which ones should not.)

**Review by Aaronontheweb — COMMENTED** [review]
Reviewed my changes

**Review by Arkatufus — APPROVED** [review]
LGTM

**Pull request merged**